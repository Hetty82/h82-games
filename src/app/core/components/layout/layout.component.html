<header class="header">
  <div class="left">
    <h1 class="title">{{ title }}</h1>
    <p>{{ subTitle }}</p>
  </div>

  <div class="center">
    <a routerLink="/login" routerLinkActive="active">Login</a>
    <a routerLink="/games" routerLinkActive="active">Games</a>
  </div>

  <div class="right">
    [ account ]
  </div>
</header>

<main class="main">
  <router-outlet></router-outlet>
</main>
