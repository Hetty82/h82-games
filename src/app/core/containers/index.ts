import { AppComponent } from './app/app.component'
import { GamesComponent } from './games/games.component'
import { LoginComponent } from './login/login.component'
import { NotFoundComponent } from './not-found/not-found.component'

export const containers: any[] = [
  AppComponent,
  GamesComponent,
  LoginComponent,
  NotFoundComponent,
]

export * from './app/app.component'
export * from './games/games.component'
export * from './login/login.component'
export * from './not-found/not-found.component'
