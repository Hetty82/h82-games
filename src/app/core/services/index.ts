import { GamesService } from './games.service'
import { UserService } from './user.service'

export const services: any[] = [
  GamesService,
  UserService,
]

export * from './games.service'
export * from './user.service'
