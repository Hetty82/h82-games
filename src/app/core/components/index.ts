import { AppComponent } from './app/app.component'
import { LayoutComponent } from './layout/layout.component'
import { NotFoundComponent } from './not-found/not-found.component'

export const components: any[] = [
  AppComponent,
  LayoutComponent,
  NotFoundComponent,
]

export * from './app/app.component'
export * from './layout/layout.component'
export * from './not-found/not-found.component'
