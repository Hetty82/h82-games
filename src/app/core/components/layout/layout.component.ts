import { Component, OnInit, Input } from '@angular/core'
import { Store, select } from '@ngrx/store'

import * as fromRoot from '../../../store'


@Component({
  selector: 'app-layout',
  templateUrl: './layout.component.html',
  styleUrls: ['./layout.component.sass']
})
export class LayoutComponent implements OnInit {
  @Input() title: string
  @Input() subTitle: string

  user$ = this.store.pipe(select(fromRoot.getCurrentUser))

  constructor(private store: Store<fromRoot.State>) {
  }

  ngOnInit() {
  }

  logout() {
    this.store.dispatch( new fromRoot.DeselectUser() )
  }
}
