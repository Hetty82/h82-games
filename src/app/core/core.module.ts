import { NgModule } from '@angular/core'
import { CommonModule } from '@angular/common'
import { RouterModule } from '@angular/router'

import * as fromComponents from './components'
import * as fromContainers from './containers'
import * as fromServices from './services'


@NgModule({
  imports: [
    CommonModule,
    RouterModule,
  ],
  declarations: [
    ...fromComponents.components,
    ...fromContainers.containers,
  ],
  exports: [
    fromContainers.AppComponent,
  ],
})
export class CoreModule {
  static forRoot() {
    return {
      ngModule: CoreModule,
      providers: fromServices.services,
    }
  }
}
