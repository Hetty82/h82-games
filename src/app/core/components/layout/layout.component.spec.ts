import { NO_ERRORS_SCHEMA } from '@angular/core'
import { TestBed, async, ComponentFixture } from '@angular/core/testing'
import { RouterTestingModule } from '@angular/router/testing'

import { LayoutComponent } from './layout.component'


describe('LayoutComponent', () => {
  let component: LayoutComponent
  let fixture: ComponentFixture<LayoutComponent>

  beforeEach(async(() => {
    TestBed.configureCompiler({ preserveWhitespaces: false } as any).configureTestingModule({
      declarations: [ LayoutComponent ],
      imports: [ RouterTestingModule ],
      schemas: [ NO_ERRORS_SCHEMA ],
    })
    .compileComponents()
  }))

  beforeEach(() => {
    fixture = TestBed.createComponent(LayoutComponent)
    component = fixture.componentInstance
    fixture.detectChanges()
  })

  it('should create component', () => {
    expect(component).toBeTruthy()
  })

  it('should match snapshot', () => {
    expect(fixture).toMatchSnapshot()
  })
})
