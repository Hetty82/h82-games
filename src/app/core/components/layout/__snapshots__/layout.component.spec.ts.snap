// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutComponent should match snapshot 1`] = `
<app-layout>
  <header /><main>
     --/ router-outlet -- 
    <router-outlet />
     -- router-outlet /--
  
  </main>
</app-layout>
`;
